### DISCUSSION TITLE
PHPstan deprecation stubs

### DISCUSSION THREAD

**Discussion by lloricode** [filamentphp/filament#5016] [Ideas]
Hi!, can we have [phpstan/phpstan-deprecation-rules](https://github.com/phpstan/phpstan-deprecation-rules) stubs for deprecation? for easy detect deprecation 

thank you

**Comment by zepfietje** [comment]
To detect deprecations in the Filament codebase? I don't think that makes sense, since some methods marked as deprecated will be kept until v4.

**Reply by lloricode** [reply]
Hi @zepfietje, 

assuming it will be remove sooner later, so deprecation will be required to update, it is possible to have filament deprecation stub? or add deprecation stub later in v4? looking forward for filament to easily detect deprecation.

I have using phpstorm so easily detect deprecation, but some many of my team they not detect deprecation, so much better if static analysis detect it

**Reply by danharrin** [reply]
We already have `@deprecated` docblocks, so I think you just need to install the rules into your own project? Not ours
